src/test/kotlin/no/nav/syfo/narmesteleder/NaermesteLederControllerTest.kt

```diff
@@ -0,0 +1,86 @@
+package no.nav.syfo.narmesteleder
+
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import no.nav.security.token.support.core.context.TokenValidationContextHolder
+import no.nav.syfo.metric.Metric
+import no.nav.syfo.narmesteleder.consumer.NaermesteLederStatus
+import no.nav.syfo.narmesteleder.consumer.Naermesteleder
+import no.nav.syfo.narmesteleder.consumer.NarmesteLederConsumer
+import no.nav.syfo.narmesteleder.controller.NaermesteLederController
+import no.nav.syfo.narmesteleder.controller.NarmesteLederMapper
+import no.nav.syfo.narmesteleder.controller.RSNaermesteLeder
+import no.nav.syfo.testhelper.OidcTestHelper.getValidationContext
+import no.nav.syfo.testhelper.UserConstants
+import no.nav.syfo.tilgang.TilgangskontrollService
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.time.LocalDate
+
+@ExtendWith(SpringExtension::class)
+class NaermesteLederControllerTest {
+    private val narmesteLederMapper: NarmesteLederMapper = mockk()
+
+    private val oidcRequestContextHolder: TokenValidationContextHolder = mockk()
+
+    private val tilgangskontrollService: TilgangskontrollService = mockk()
+
+    private val narmesteLederConsumer: NarmesteLederConsumer = mockk()
+
+    private val metric: Metric = mockk()
+
+    private val naermestelederController = NaermesteLederController(
+            metric,
+            oidcRequestContextHolder,
+            tilgangskontrollService,
+            narmesteLederMapper,
+            narmesteLederConsumer
+    )
+
+    @Test
+    fun hentNaermesteLederSuccess() {
```

Comment:
Jeg synes denne testen og tilsvarende test for NarmesteLedereController er litt i tynneste laget. Hva er det egentlig de tester? De var nok allerede tynn fra før endringene, men ettersom mapping er dratt ut, så tester den enda mindre. Egentlig burde dette kanskje løses med en integrasjonstest? Det har vi ikke snakket så mye om når det gjelder testing, og jeg er usikker på om vi har noe særlig integrasjonstester i det hele tatt. Dersom vi ønsker å beholde den som enhetstest så foreslår jeg å lage flere tester, der man tester bl.a. tilfeller der man ikke finner en nærmeste leder, der den nærmeste lederen ikke er aktiv eller at man prøver å hente for et fnr man ikke har tilgang til.

Siden mapping er tatt ut i en egen klasse, så bør den kanskje også få sin egen unittest?

Reply:
Veldig bra poeng ang flere test-caser, det har jeg fikset.
Når det gjelder unittest for mapper: blir det noen verdi for den? Mapperen har jo "dum" og inneholder ingen logikk.